src/main/java/seedu/address/model/customer/BudgetAndOneTagsPredicate.java

```diff
@@ -26,8 +26,12 @@ public BudgetAndOneTagsPredicate(Budget budget, Set<Tag> tags) {
 
     @Override
     public boolean test(Customer customer) {
-        return tags.stream().anyMatch(tag -> customer.getTags().contains(tag))
-                && customer.getBudget().isInRangeBudget(budget);
+        if (tags.size() == 0) {
+            return customer.getBudget().isInRangeBudget(budget);
+        } else {
+            return tags.stream().anyMatch(tag -> customer.getTags().contains(tag))
+                    && customer.getBudget().isInRangeBudget(budget);
+        }
```

Comment:
I think you can simplify it to be like this:

```suggestion
        return (tags.size() == 0 || tags.stream().anyMatch(tag -> customer.getTags().contains(tag)))
            && customer.getBudget().isInRangeBudget(budget);
```